src/canister_tests/src/tests.rs

```diff
@@ -52,3 +60,24 @@ fn ii_canister_can_be_upgraded_and_rolled_back() {
     framework::upgrade_ii_canister(&env, canister_id, framework::II_WASM_PREVIOUS.clone());
     api::health_check(&env, canister_id);
 }
+
+#[test]
+fn registration_with_mismatched_sender_fails() {
+    let env = StateMachine::new();
+    let canister_id = framework::install_ii_canister(&env, framework::II_WASM_PREVIOUS.clone());
+    let challenge = api::create_challenge(&env, canister_id);
+    let result = register(
+        &env,
+        canister_id,
+        principal_2(),
+        device_data_1(),
```

Comment:
I think this improves readability + understanding a lot, wdyt?

Reply:
Yes, agreed 👍